### PR TITLE
frontend: Add a dedicated CustomResource details view

### PR DESCRIPTION
**Pull request by joaquimrocha**
We were showing custom resources only with a YAML viewer and this
didn't allow us to edit/remove a custom resource. This patch adds a
simple custom resource details view which allows to perform the
mentioned operations.
fixes #367

### CODE REVIEW

File: frontend/package.json
Comment (ashu8912) [inline frontend/package.json:45]: can we move **package.json** and **package-lock.json** to a different commit explaining why we need this dependency?

File: frontend/src/components/crd/ResourceDetails.tsx
Comment (ashu8912) [reply frontend/src/components/crd/ResourceDetails.tsx:14]: @joaquimrocha I think it was agreed that for each new component we add we also add test/story for it, if you are busy with things maybe I can add a story for this PR??

File: frontend/src/components/crd/ResourceDetails.tsx
Comment (ashu8912) [inline frontend/src/components/crd/ResourceDetails.tsx:48]: same as earlier this won't be required if we add a story for CRD detail view

File: frontend/src/components/crd/ResourceDetails.tsx
Comment (ashu8912) [inline frontend/src/components/crd/ResourceDetails.tsx:49]: handle coverage

File: frontend/src/components/crd/ResourceDetails.tsx
Comment (ashu8912) [inline frontend/src/components/crd/ResourceDetails.tsx:57]: handle coverage

File: frontend/src/components/crd/ResourceDetails.tsx
Comment (ashu8912) [inline frontend/src/components/crd/ResourceDetails.tsx:95]: handle coverage

**Comment by ashu8912** [comment]
I have tested the PR and it LGTM!!, after above comments are addressed we can merge it

File: frontend/src/components/crd/ResourceDetails.tsx
Comment (joaquimrocha) [reply frontend/src/components/crd/ResourceDetails.tsx:14]: I have briefly discussed this with @illume . It's very difficult to test components that use API hooks, and many components actually use another so tests would be 90% the same. We should automate this somehow. I will open an issue.

File: frontend/package.json
Comment (joaquimrocha) [reply frontend/package.json:45]: Dependencies shouldn't go in a different commit. Otherwise a bisect may fail.

File: frontend/src/components/crd/ResourceDetails.tsx
Comment (joaquimrocha) [reply frontend/src/components/crd/ResourceDetails.tsx:48]: Why wouldn't this be required if we had a story?

File: frontend/src/components/crd/ResourceDetails.tsx
Comment (illume) [reply frontend/src/components/crd/ResourceDetails.tsx:14]: I'm looking at making a story for the component in this PR now.

**Comment by illume** [comment]
To see CRD in Headlamp:

> install inspektor gadget + the seccomp profile stuff. Docs on [IG](https://github.com/kinvolk/inspektor-gadget)

File: frontend/src/components/crd/CustomResourceDetails.tsx
Comment (joaquimrocha) [inline frontend/src/components/crd/CustomResourceDetails.tsx:39]: I think this would constitue using a hook conditionally (not allowed in React), and the reason why I had two components.

**Comment by joaquimrocha** [comment]
I have updated the PR with a more creative way to test the related components at a much a higher percentage. Hopefully this is okay now. Please give it another go, @ashu8912 .

**Review by ashu8912 — APPROVED** [review]

**Pull request merged**
